packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -439,14 +456,16 @@ export default function () {
           }
 
           // add __esModule declaration if this file has any exports
-          if (hasExports && !strict) {
```

Comment:
Yeah this argument is poorly named, this is `strict` module transpiling, e.g. without adding a magic `__esModule` export. `strictMode` is the correct flag these days to disable strict mode in modules.